rest_util/capool.go

```diff
@@ -0,0 +1,88 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package rest_util
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"github.com/fullsailor/pkcs7"
+	"io/ioutil"
+)
+
+// VerifyController will attempt to use the provided x509.CertPool to connect to the provided controller.
+// If successful true an no error will be returned.
+func VerifyController(controllerAddr string, caPool *x509.CertPool) (bool, error) {
```

Comment:
Just an unecessary nit picky question. Will there one day be a scenario where we can expect a `false, nil` return? Otherwise seems a bit redundant to be checking the bool here if they always mean the same thing

Reply:
`false` is the zero value of a bool - so there is no way to both return true/false and or nil/not nill errors w/o it being false.

The scenarios are:

- `true, nil` - all is okay
- `true, error` - not possible
- `false, error` - there was a connection error or other runtime error, could not validate the controller and here is the error that cause the issue
- `false, nil` - the controller was contacted and failed to validate, there was no unexpected errors